src/ucp/rndv/rndv.c

```diff
@@ -544,33 +547,51 @@ ucp_rndv_progress_rma_zcopy_common(ucp_request_t *req, ucp_lane_index_t lane,
     }
 }
 
+static UCS_F_ALWAYS_INLINE ucp_request_t*
+ucp_rndv_rma_get_zcopy_req(ucp_request_t *req);
+
 UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_rma_get_zcopy, (self),
                  uct_pending_req_t *self)
 {
-    ucp_request_t *rndv_req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_t *rndv_req;
```

Comment:
is changes in this func related to the reported issue (or) covering some other issue?

Reply:
yes, it fixes issue appeared after fixing the original one.
normal GET Zcopy fallbacks to pipeline GET Zcopy, but it couldn't be done too, so it should send RTR. But need to correctly release all things allocated for GET.

Reply:
when is the pipeline GET ZCOPY can fail? it does GET on HOST memory which should always be a success with both IB and TCP.   is it because you explicitly force disabled UCT_IFACE_FLAG_GET_ZCOPY in the gtest?

Reply:
> when is the pipeline GET ZCOPY can fail? it does GET on HOST memory which should always be a success with both IB and TCP. is it because you explicitly force disabled UCT_IFACE_FLAG_GET_ZCOPY in the gtest?

yes, when there is no `UCT_IFACE_FLAG_GET_ZCOPY` capability on transports.

Reply:
in that case,  isn't '`"is_get_zcopy_supported`" is false and `ucp_rndv_recv_start_get_pipeline`  is not called?

Reply:
> in that case, isn't '`"is_get_zcopy_supported`" is false and `ucp_rndv_recv_start_get_pipeline` is not called?,

if `is_get_zcopy_supported` - `false`, then `ucp_rndv_recv_start_get_pipeline` is not called, since GET operation can't be done

Reply:
@bureddy the changes is needed to release resources (mdesc, request) for fragmented GET protocol if it couldn't be used by some reason (i.e. `ucp_rndv_zcopy_get_lane()` returns `UCP_NULL_LANE`).